src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs

```diff
@@ -1130,7 +1130,7 @@ public static void Equal(Half expected, Half actual, Half variance)
         /// <exception cref="EqualException">Thrown when the representations are not identical</exception>
         public static void Equal(double expected, double actual)
         {
-            if (BitConverter.ToInt64(BitConverter.GetBytes(expected)) == BitConverter.ToInt64(BitConverter.GetBytes(actual)))
+            if (BitConverter.ToInt64(BitConverter.GetBytes(expected), 0) == BitConverter.ToInt64(BitConverter.GetBytes(actual), 0))
```

Comment:
This should be `BitConverter.DoubleToInt64Bits(expected)`

Reply:
and similarly for the ones further down. For `float` on .NET Framework, I'd just define a helper:
```csharp
static unsafe int SingleToInt32Bits(float value)
{
    return *(int*)&value;
}
```

Reply:
Done, thank you for suggestion. By the way I didn't find anything in the supported product list for the ```BitConverter.HalfToInt16Bits``` function in [.NET API Catalog](https://apisof.net/catalog/94ba24bd-01db-1896-4a9c-bfdff6ad8a5b) except the .NET version

Reply:
The official API docs include the .NET versions its been included in: https://learn.microsoft.com/en-us/dotnet/api/system.bitconverter.halftoint16bits#applies-to